src/core/theme/index.ts

```diff
@@ -15,13 +15,22 @@ export type SpacingProp = keyof typeof importedTokens.spacing;
 export type SuomifiTheme = ReturnType<typeof suomifiTheme>;
 
 export interface TokensProp {
-  tokens?: SuomifiTokens;
+  /** Custom  Design-tokens or one category of tokens customized, clone defaultTokens for base */
+  tokens?: Partial<SuomifiTokens>;
+}
+
+export interface DefinedTokensProp {
```

Comment:
This naming feels confusing to me. What is the meaning of "Defined" in this context?

Reply:
If I understood correctly this typing is used for the resolved tokens that are the result of merge of default and user provided tokens. => "Defined" simply means that all the tokens are present. In addition this is only used internally and not a part of the public API.

Could we use a name that describes all three properties: internal nature, non optional existence and the merge from default and custom tokens? Say ResolvedTokensProp?

Reply:
Even if ResolvedTokensProp describes the process used to come up with data of this type, it also describes what it is used for and implies that it is defined as internally needed (resolved).